drivers/i2c/i2c_gecko.c

```diff
@@ -37,6 +37,7 @@ struct i2c_gecko_config {
 
 struct i2c_gecko_data {
 	struct k_sem device_sync_sem;
+	struct k_sem bus_lock;
```

Comment:
I believe it should be a `k_mutex` rather than a `k_sem`.

Reply:
In my initial commit, the variable was inconsistent (mutex), but the semaphore behavior is correct.
Change variable name from bus_mutex to bus_lock

Reply:
Yep, and `k_sem` is almost always a better choice than `k_mutex` when it comes to locking. I recommend checking out https://www.youtube.com/watch?v=vla8DTsttyA by @andyross which has lots of interesting insights regarding this.